File: Easy/Fraction Trouble.cpp
//{ Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++
class Solution
{
  public:
    vector<int> numAndDen(int n, int d)
    {
        // code here
        /*
        
            You have to find out p/q where, 
            p/q < n/d
            pd < qn
            pd <= qn-1 .
        
        */
        
        int num = -1;
        int den = 1;
        
        for (int i=d+1; i < 10001; i++)
        {
            int j = (n*i)/d;
            if (__gcd(j,i) == 1) 
            {
                if (1.0 * j/i > 1.0 * num/den)
                {
                    num = j;
                    den = i;
                }
        }
    }
    
    return {num, den};
        
        
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n,d;
        cin >> n >> d;
        Solution ob;
        vector<int>ans=ob.numAndDen(n,d);
        for(int i=0;i<ans.size();i++)cout<<ans[i]<<" ";
        cout<<endl;
    }
    return 0;
}

// } Driver Code Ends

File: Easy/Fraction Trouble.py
#User function Template for python3
import math

class Solution:
	def LargestFraction(self, n, d):
		# Code here
		numeratore = 0
		denominatore = 1
		
		div = 0
		
		m = 10000
		
		while m >= 2:
		    p = (n * m - 1) // d
		    if p * denominatore >= m * numeratore:
		        numeratore = p
		        denominatore = m
		    m -= 1
		    
	    div = math.gcd(numeratore, denominatore)
	    
	    return numeratore // div, denominatore // div


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n, d = input().split()
		n = int(n); d = int(d);
		ob = Solution()
		ans = ob.LargestFraction(n, d)
		for _ in ans:
			print(_, end = " ")
		print()
# } Driver Code Ends